cpp/harvest/marc_issn_lookup.cc

```diff
@@ -0,0 +1,171 @@
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <fstream>
+#include <iostream>
+#include "CORE.h"
+#include "FileUtil.h"
+#include "MARC.h"
+#include "util.h"
+
+namespace {
+[[noreturn]] void Usage() {
+    ::Usage(
+        "marc_input_articles marc_input_journals marc_output_articles"
+        "\n"
+        "- marc_input_artiles is a file containing all article information taken from CORE. \n"
+        "- marc_input_journals is a file containing journal information. Please use issn_lookup.py to generate this file. \n"
+        "- marc_output_articles is an output file generated by this tool. \n");
+
+    std::exit(EXIT_FAILURE);
+}
+struct SubFieldInfo {
+    std::string i;
+    std::string t;
+    std::string w;
+    std::string x;
+    bool is_found;
+
+    SubFieldInfo(bool b) { this->is_found = b; }
+
+    void Builder(MARC::Record &record) {
+        for (auto &field : record) {
+            if (field.getTag() == "001")
+                this->w = "(DE-627)" + field.getContents();
+
+            if (field.getTag() == "773")
+                this->x = field.getContents();
+
+            if (field.getTag() == "245") {
+                MARC::Subfields subfields(field.getSubfields());
+                std::string subfield_a(subfields.getFirstSubfieldWithCode('a'));
+                std::string subfield_b(subfields.getFirstSubfieldWithCode('b'));
+                if (subfield_a != "" && subfield_b != "")
+                    this->t = subfield_a + " " + subfield_b;
+                else if (subfield_a != "" && subfield_b == "")
+                    this->t = subfield_a;
+                else if (subfield_a == "" && subfield_b != "")
+                    this->t = subfield_b;
+                else
+                    this->t = "";
+            }
+        }
+        this->i = "In:";
+        this->is_found = true;
+    }
+};
+
+bool IsOnlineISSN(const MARC::Record &record) {
+    for (auto &field : record) {
+        if (field.getTag() == "300") {
+            const std::string sub_field_a(field.getFirstSubfieldWithCode('a'));
+            if (sub_field_a == "Online-Ressource")
+                return true;
+            else
+                return false;
+        }
+    }
+    return false;
+}
+
+SubFieldInfo LookupISSNInDataSouce(const std::string input_journal_filename, const std::string issn) {
```

Comment:
What you currently do: Find the first journal for this ISSN, check whether it's print or online, and return the corresponding information.

BUT: You can have 10 journals for the same ISSN, and the 7th could be the online journal.
So that's what i meant: you have to put all journal candidates into a set and then check it in the corresponding priority:
1) Return the online journal if you have exact 1
2) Return a print journal if you have exact 1
3) return nothing (is_found = false)